addons/addon-base-raas/packages/base-raas-post-deployment/lib/steps/create-service-catalog-portfolio.js

```diff
@@ -295,13 +300,35 @@ class CreateServiceCatalogPortfolio extends Service {
         DisableTemplateValidation: true,
         Type: 'CLOUD_FORMATION_TEMPLATE',
         Description: productToCreate.description || autoCreateDesc,
+        Name: nextProvArtifact,
       },
       ProductId: productId,
     };
     const data = await servicecatalog.createProvisioningArtifact(params).promise();
     return data.ProvisioningArtifactDetail.Id;
   }
 
+  async getNextArtifactVersion(latestVersionInSC) {
+    // For product artifacts in Service Catalog, versions and names are the same thing
+    // If no previous version found, assign 'v2' (this will help in next update cycle)
+    let returnVal = 'v2';
+    // Only finds version strings that match patterns:
+    // "V<n>", "V<n>.0", "V<n>.0.0", "v<n>", "v<n>.0", "v<n>.0.0"
+    const pattern = /^(v|V)(\d+\.)?(\d+\.)?(\*|\d+)$/;
+    let latestVersionNo;
+    if (latestVersionInSC) {
+      const parsedOutput = latestVersionInSC.match(pattern);
+      if (parsedOutput && parsedOutput.length > 1) {
+        const longestMatch = parsedOutput.sort(function(a, b) {
+          return b.length - a.length;
+        })[0];
+        latestVersionNo = longestMatch.substring(1);
+        returnVal = `v${parseInt(latestVersionNo.split('.')[0], 10) + 1}`;
```

Comment:
Using regex AND string methods (i.e. substring, split) together is often an anti pattern. It leads to fragmented/duplicated string parsing logic. 

A regex with well placed capture groups can find the version number you are looking for. 

Try something like
```regex
^(?:v|V)(\d+)?(?:\.\d+)?(?:\.(?:\*|\d+))?$
```
https://regexr.com/5e53g

Reply:
Made the changes per our conversation. Thanks for this suggestion 👍